src/com/sap/piper/Notify.groovy

```diff
@@ -0,0 +1,51 @@
+package com.sap.piper
+
+import groovy.text.SimpleTemplateEngine
+
+class Notify implements Serializable {
+    private static enum Severity { ERROR, WARNING }
+    private final static String LIBRARY_NAME = 'piper-lib-os'
+    private final static String MESSAGE_PATTERN = '[${severity}] ${message} (${libName}/${stepName})'
+
+    protected static Utils instance = null
+
+    protected static Utils getUtilsInstance(){
+        instance = instance ?: new Utils()
+        return instance
+    }
+
+    static void warning(Map config, Script step, String msg, String stepName = null){
+        log(config, step, msg, stepName)
+    }
+
+    static void error(Map config, Script step, String msg, String stepName = null) {
+        log(config, step, msg, stepName, Severity.ERROR)
+    }
+
+    private static void log(Map config, Script step, String msg, String stepName, Severity severity = Severity.WARNING){
+        stepName = stepName ?: step.STEP_NAME
+        getUtilsInstance().pushToSWA([
+            folder: '',
+            repository: '',
+            step: 'Notify',
+            eventType: 'notification',
+            stepParam1: LIBRARY_NAME,
+            stepParam2: stepName,
+            stepParam3: msg,
+            stepParam4: severity
+        ], config)
+        def logEntry = SimpleTemplateEngine.newInstance().createTemplate(
+            MESSAGE_PATTERN
+        ).make([
+            libName: LIBRARY_NAME,
+            stepName: stepName,
+            message: msg,
+            severity: severity
+        ]).toString()
+
+        if (severity == Severity.ERROR){
```

Comment:
Not sure if it is a cool idea to bundle to different aspects (abort execution by raising an exception and logging) in one step. IMO it i better to only log here and not throwing exceptions. That I would also not expect from a `Notify` class. IMO bundling several aspects in one class is never a cool thing. Basically another incarnation of the good old linux principle ["one tool, one job"](https://en.wikipedia.org/wiki/Unix_philosophy#Do_One_Thing_and_Do_It_Well).

Reply:
This is not about logging, it's about raise pipeline issue in a standard format that they can be picked up by a log crawler and displayed as issues in the Jenkins ui, see #652.

Reply:
I'm not convinced that `Notify.error` should throw an exception. What we have here mixes  basically several aspects:

1.) logging
2.) sending notification 
3.) raising an exception.

Having several aspects in one makes life more complicated than it needs to be. What if I would like to write an error (with a unified format) in the log, but I would like not to fail with an exception?

What when I would like to emit a warning (with a unified format) in the log, but I don't want to send a notification?